### PR TITLE
fix: mac signing issue with arm64

### PR DESCRIPTION
**Pull request by sspaink**
The arm64 binary failed to get signed during the release because by default Go gives an ad hoc signature to arm64 binaries because Mac requires a signed binary with arm64 otherwise all Go binaries will fail (thread about this here: https://github.com/golang/go/issues/42684). Therefore in order to overwrite the ad hoc sig with our influxdata signature we needed to provide the flag `--force` to the codesign tool. This pull requests adds the required `--force` flag and refactored the mac signing shell script. All linter issues found by shellcheck were addressed, more information is printed out for future debugging, and I updated the script to stop changing directories and instead use absolute paths to make it easier to read.

This was tested by changing the circleci config to only build and sign the mac artifacts, you can see the results here: 
- https://app.circleci.com/pipelines/github/influxdata/telegraf/8166/workflows/f51a71e7-3a9d-4311-b3a0-6c4e9df4071e 

you can find the artifacts that include the signed dmg files here: 
- https://app.circleci.com/pipelines/github/influxdata/telegraf/8166/workflows/f51a71e7-3a9d-4311-b3a0-6c4e9df4071e/jobs/138696/artifacts

### CODE REVIEW

**Review by MyaLongmire — APPROVED** [review]
Thank you so much for figuring this out!

File: scripts/mac-signing.sh [hunk 1]
```diff
@@ -1,79 +1,101 @@
...
+armFile=$(find "$HOME/project/dist" -name "*darwin_arm64.tar*")
+macFiles=("${amdFile}" "${armFile}")
+
+for tarFile in "${macFiles[@]}";
+do
+  cleanup
+
+  # Create the .app bundle directory structure
+  mkdir -p Telegraf.app/Contents
+  RootAppDir="Telegraf.app/Contents"
```
Comment (powersj) [inline scripts/mac-signing.sh:30]:
```suggestion
  RootAppDir="Telegraf.app/Contents"
  mkdir -p "$RootAppDir"
```

File: scripts/mac-signing.sh [hunk 2]
```diff
@@ -1,79 +1,101 @@
...
   # Attach the notarization to the DMG.
   xcrun stapler staple "$baseName".dmg
-  rm -rf Telegraf.app
-  rm -rf $extractedFolder
-  ls
+  cleanup
+
+  mv "$baseName".dmg ~/project/dist
 
   echo "$tarFile Signed and notarized!"
```
Comment (powersj) [inline scripts/mac-signing.sh:100]:
```suggestion
  echo "$baseName.dmg signed and notarized!"
```

**Review by powersj — COMMENTED** [review]
Awesome work and a big thank you for the clean-up. This should make this easier in the future if we ever need to debug any issues! Two small changes, otherwise this is good to go.

**Review by powersj — APPROVED** [review]

**Pull request merged**